### ISSUE TITLE
A use-case of semi-supervised learning for images

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/neural-structured-learning#104]
@arjung since semi-supervision has come a long way in computer vision. Since Neural Structured Learning has first-class support to perform semi-supervision I was thinking of example doing the following:

* Take the 10000 images randomly from CIFAR10 and also split the rest of the samples into unlabeled and validation samples.
* Train a model on this subset. This model will be some pre-trained ImageNet-1k model fine-tuned on the CIFAR10 10000 samples I mentioned. 
* Use this model (without the final classification layer) as a feature extractor to generate embeddings for the subset and rest of the 40000 images in CIFAR10.
* Build the graph augmented with structural similarity.
* Train another model with this graph data.
* Evaluate the model on the test set.

What do you think of this?
